# NB92 — The Gravitational Propagator: Why No $8\pi$

## Target: Show that $G_N = 1/M_{Pl}^2$ emerges from the solenoid without the $8\pi$ factor

In standard GR, Newton's constant relates to the **reduced** Planck mass:
$$G_N = \frac{1}{8\pi M_{Pl,\text{red}}^2}$$

The $8\pi$ comes from two sources:
- **$4\pi$** from Gauss's law in 3D: $\oint \mathbf{g} \cdot d\mathbf{A} = -4\pi GM$ (the solid angle of $S^2$)
- **Factor 2** from the trace-reversal relating the Einstein tensor to the stress-energy tensor

The framework uses the **non-reduced** Planck mass: $G_N = 1/M_{Pl}^2$, with no $8\pi$.
This notebook asks: **why?**

### The argument

1. The solenoid configuration space is **flat** (NB85) — Riemann tensor vanishes identically
2. There is no Einstein-Hilbert action, no Ricci scalar, no Einstein equations
3. The $8\pi$ has no origin because the mathematical structures that produce it don't exist
4. Gravity on the solenoid is a **spectral-metric propagation** phenomenon:
   the gravitational coupling is the metric propagator $(g^{-1})^{\omega-1}_{1,\omega} = -1/\lambda(P_4)$
5. The Gauss's law analog on the solenoid replaces the continuous solid angle $4\pi$
   with the discrete group structure $\phi(P_4) = 48$

### Plan

1. **Section 1**: The 8π in standard physics — anatomy
2. **Section 2**: The solenoid metric propagator
3. **Section 3**: Discrete vs continuous Gauss's law
4. **Section 4**: Dimensional reduction — why no geometric prefactor
5. **Section 5**: The solenoid's gravitational coupling constant
6. **Section 6**: Scorecard

In [1]:
# -- NB92 Setup --
import sys, numpy as np, math, sympy
from pathlib import Path
from fractions import Fraction

ROOT = Path.cwd().parent
if str(ROOT / 'scripts') not in sys.path:
    sys.path.insert(0, str(ROOT / 'scripts'))

from solenoid_algebra import (SA, RHO, KAPPA, EPSILON, OMEGA,
                               PRIMES, P, PHI, GROUP_EXPONENT, PRIMORIALS)

# Physical constants
M_Z  = 91.1876         # GeV
M_Pl = 1.22089e19      # GeV (non-reduced)
M_Pl_red = M_Pl / np.sqrt(8 * np.pi)  # reduced Planck mass
G_N_CODATA = 6.67430e-11  # m^3 kg^-1 s^-2
G_N_natural = 1 / M_Pl**2  # GeV^-2 (non-reduced)
G_N_reduced = 1 / (8 * np.pi * M_Pl_red**2)  # should equal G_N_natural

# Hierarchy
H_sol = 240**4 * 7**9

print("NB92 -- THE GRAVITATIONAL PROPAGATOR: WHY NO 8pi")
print("=" * 65)
print(f"  M_Pl (non-reduced) = {M_Pl:.5e} GeV")
print(f"  M_Pl (reduced)     = {M_Pl_red:.5e} GeV")
print(f"  Ratio M_Pl/M_Pl_red = sqrt(8pi) = {M_Pl/M_Pl_red:.6f}")
print(f"  sqrt(8pi)           = {np.sqrt(8*np.pi):.6f}")
print(f"  8pi = {8*np.pi:.6f}")
print(f"\n  G_N = 1/M_Pl^2 = {G_N_natural:.5e} GeV^-2")
print(f"  Check: 1/(8pi M_Pl_red^2) = {G_N_reduced:.5e} GeV^-2")
print(f"  Match: {abs(G_N_natural/G_N_reduced - 1) < 1e-10}")

NB92 -- THE GRAVITATIONAL PROPAGATOR: WHY NO 8pi
  M_Pl (non-reduced) = 1.22089e+19 GeV
  M_Pl (reduced)     = 2.43532e+18 GeV
  Ratio M_Pl/M_Pl_red = sqrt(8pi) = 5.013257
  sqrt(8pi)           = 5.013257
  8pi = 25.132741

  G_N = 1/M_Pl^2 = 6.70883e-39 GeV^-2
  Check: 1/(8pi M_Pl_red^2) = 6.70883e-39 GeV^-2
  Match: True


## Section 1: Anatomy of the $8\pi$ in Standard Physics

The factor $8\pi$ in $G_N = 1/(8\pi M_{Pl,\text{red}}^2)$ arises from two independent sources:

**Source 1: The $4\pi$ from Gauss's Law (3D geometry)**

In Newtonian gravity, the Poisson equation is:
$$\nabla^2 \Phi = 4\pi G \rho$$

The $4\pi$ is the solid angle of the 2-sphere $S^2$ — the surface at constant radius in 3D space.
It arises because in 3D, the gravitational flux through a closed surface is:
$$\oint_{\partial V} \mathbf{g} \cdot d\mathbf{A} = -4\pi G M_{enc}$$

The $4\pi$ is purely a property of the **dimensionality** of space:
- In $d$ dimensions: the solid angle is $\Omega_d = 2\pi^{d/2} / \Gamma(d/2)$
- For $d=3$: $\Omega_3 = 4\pi$
- For $d=4$: $\Omega_4 = 2\pi^2$

**Source 2: The factor 2 from Einstein's trace-reversal**

The Einstein equations $G_{\mu\nu} = 8\pi G T_{\mu\nu}$ relate geometry to matter.
The extra factor of 2 (beyond the $4\pi$ from Gauss's law) comes from the trace-reversal:
$$R_{\mu\nu} - \frac{1}{2}g_{\mu\nu}R = 8\pi G T_{\mu\nu}$$

The $-\frac{1}{2}g_{\mu\nu}R$ term, when contracted to give the weak-field Poisson equation,
contributes an additional factor of 2.

**Both sources are Cartesian artifacts**:
- The $4\pi$ requires 3D spatial geometry with a radial coordinate
- The trace-reversal requires a Lorentzian metric with 3+1 signature
- Neither structure exists on the solenoid

In [2]:
# Section 1: Anatomy of 8pi
print("SECTION 1: ANATOMY OF 8pi IN STANDARD PHYSICS")
print("=" * 65)

# The 8pi decomposition
print(f"  8pi = 4pi x 2 = {8*np.pi:.6f}")
print(f"       = (3D solid angle) x (trace-reversal factor)")

# Solid angles in various dimensions
from math import gamma as gamma_func
print(f"\n  Solid angles by dimension:")
print(f"  {'d':>4} {'Omega_d':>14} {'Formula':>20}")
print(f"  {'-'*42}")
for d in range(1, 8):
    omega_d = 2 * np.pi**(d/2) / gamma_func(d/2)
    formula = f"2pi^({d}/2)/Gamma({d}/2)"
    marker = " <-- our 3D" if d == 3 else ""
    print(f"  {d:>4} {omega_d:>14.6f} {formula:>20}{marker}")

# On the solenoid: there are no spatial dimensions in the Cartesian sense
print(f"\n  ON THE SOLENOID:")
print(f"    Configuration space: 5D (t, R1, R2, R3, R4)")
print(f"    Metric: constant (all Christoffel symbols = 0)")
print(f"    Riemann tensor: identically zero (NB85)")
print(f"    There is no Ricci scalar R")
print(f"    There is no Einstein-Hilbert action")
print(f"    There is no Poisson equation with nabla^2")
print(f"    Therefore: 4pi has no origin")
print(f"    Therefore: the trace-reversal factor 2 has no origin")
print(f"    Therefore: 8pi has no origin")

SECTION 1: ANATOMY OF 8pi IN STANDARD PHYSICS
  8pi = 4pi x 2 = 25.132741
       = (3D solid angle) x (trace-reversal factor)

  Solid angles by dimension:
     d        Omega_d              Formula
  ------------------------------------------
     1       2.000000 2pi^(1/2)/Gamma(1/2)
     2       6.283185 2pi^(2/2)/Gamma(2/2)
     3      12.566371 2pi^(3/2)/Gamma(3/2) <-- our 3D
     4      19.739209 2pi^(4/2)/Gamma(4/2)
     5      26.318945 2pi^(5/2)/Gamma(5/2)
     6      31.006277 2pi^(6/2)/Gamma(6/2)
     7      33.073362 2pi^(7/2)/Gamma(7/2)

  ON THE SOLENOID:
    Configuration space: 5D (t, R1, R2, R3, R4)
    Metric: constant (all Christoffel symbols = 0)
    Riemann tensor: identically zero (NB85)
    There is no Ricci scalar R
    There is no Einstein-Hilbert action
    There is no Poisson equation with nabla^2
    Therefore: 4pi has no origin
    Therefore: the trace-reversal factor 2 has no origin
    Therefore: 8pi has no origin


## Section 2: The Solenoid Metric Propagator

On the solenoid, gravity is not curvature. Configuration space is flat($NB85$).
Instead, the gravitational coupling emerges from the **metric propagator** —
the inverse metric $\tilde{g}_R^{-1}$ raised to powers.

NB89 discovered that the metric propagator from the innermost level ($p_1 = 2$)
to the outermost level ($p_4 = 7$) is:
$$(\tilde{g}_R^{-1})^{\omega-1}_{1,\omega} = -\frac{1}{\lambda(P_4)} = -\frac{1}{12}$$

This is identity #206 — it holds exactly and fails for ALL proper sub-blocks.

The gravitational hierarchy $M_{Pl}/M_Z = Tr(L)^{\omega} \times p_4^{d_1^2} = 240^4 \times 7^9$
is the product of:
- The **spectral scale** $Tr(L) = 240$ raised to $\omega(P_4) = 4$ (the rank of $\tilde{g}_R^{-1}$)
- The **bridge-break prime** $p_4 = 7$ raised to $d_1^2 = 9 = \sigma_3(p_1)$ (the metric innermost diagonal squared)

The coupling $1/H^2$ emerges from the metric propagator multiplied by the spectral
content of the Cayley Laplacian. There is no separate "geometric prefactor" —
the geometry IS the coupling.

In [3]:
# Section 2: The solenoid metric propagator
print("SECTION 2: THE SOLENOID METRIC PROPAGATOR")
print("=" * 65)

# Build the Schur complement inverse metric g_schur^{-1} from NB83
# Diagonal: (1+p_k)/P_{k-1} for k=1..4
# Sub-diagonal: -1/P_{k-2} for k=2..4
primes = PRIMES
prim = [1] + list(PRIMORIALS)  # P_0=1, P_1=2, P_2=6, P_3=30, P_4=210

omega = len(primes)  # 4

# Build the 4x4 Schur complement inverse (spatial part of g^{-1})
g_inv = np.zeros((omega, omega), dtype=object)
for i in range(omega):
    g_inv[i, i] = Fraction(1 + primes[i], prim[i])
    if i > 0:
        g_inv[i, i-1] = Fraction(-1, prim[i-1])
        g_inv[i-1, i] = Fraction(-1, prim[i-1])

print("  Schur complement inverse metric g_R^{-1} (4x4):")
for i in range(omega):
    row = [str(g_inv[i, j]) for j in range(omega)]
    print(f"    {row}")

# Trace and determinant
tr = sum(g_inv[i, i] for i in range(omega))
print(f"\n  Trace = {tr} = {float(tr):.6f}")

# Compute det using sympy for exact arithmetic
g_inv_sym = sympy.Matrix(omega, omega, lambda i, j: sympy.Rational(g_inv[i, j].numerator, g_inv[i, j].denominator))
det_val = g_inv_sym.det()
print(f"  Det = {det_val} = {float(det_val):.6f}")
print(f"  1/Det = {1/det_val} = {float(1/det_val):.6f}")
print(f"  Expect 179/180: {Fraction(179, 180)} = {179/180:.6f}")

# Now compute powers of g_inv
print(f"\n  METRIC PROPAGATOR: powers of g_R^{{-1}}")
print(f"  The (1,omega) entry of (g_R^{{-1}})^n connects level 1 to level omega")
g_power = sympy.eye(omega)
for n in range(1, omega+1):
    g_power = g_power * g_inv_sym
    val_1_omega = g_power[0, omega-1]
    print(f"    n={n}: (g_R^{{-1}})^{n}[1,{omega}] = {val_1_omega} = {float(val_1_omega):.8f}")

# The critical identity
prop_val = (g_inv_sym**3)[0, omega-1]
lambda_P4 = GROUP_EXPONENT
print(f"\n  IDENTITY #206:")
print(f"    (g_R^{{-1}})^(omega-1)[1,omega] = (g_R^{{-1}})^3[1,4]")
print(f"    = {prop_val}")
print(f"    -1/lambda(P4) = -1/{lambda_P4} = {Fraction(-1, lambda_P4)}")
print(f"    Match: {prop_val == sympy.Rational(-1, lambda_P4)}")

print(f"\n  This propagator connects the INNERMOST level (p=2)")
print(f"  to the OUTERMOST level (p=7) in omega-1 = 3 hops.")
print(f"  It is the solenoid's GRAVITATIONAL COUPLING CONSTANT.")

SECTION 2: THE SOLENOID METRIC PROPAGATOR
  Schur complement inverse metric g_R^{-1} (4x4):
    ['3', '-1', '0', '0']
    ['-1', '2', '-1/2', '0']
    ['0', '-1/2', '1', '-1/6']
    ['0', '0', '-1/6', '4/15']

  Trace = 94/15 = 6.266667
  Det = 179/180 = 0.994444
  1/Det = 180/179 = 1.005587
  Expect 179/180: 179/180 = 0.994444

  METRIC PROPAGATOR: powers of g_R^{-1}
  The (1,omega) entry of (g_R^{-1})^n connects level 1 to level omega
    n=1: (g_R^{-1})^1[1,4] = 0 = 0.00000000
    n=2: (g_R^{-1})^2[1,4] = 0 = 0.00000000
    n=3: (g_R^{-1})^3[1,4] = -1/12 = -0.08333333
    n=4: (g_R^{-1})^4[1,4] = -47/90 = -0.52222222

  IDENTITY #206:
    (g_R^{-1})^(omega-1)[1,omega] = (g_R^{-1})^3[1,4]
    = -1/12
    -1/lambda(P4) = -1/12 = -1/12
    Match: True

  This propagator connects the INNERMOST level (p=2)
  to the OUTERMOST level (p=7) in omega-1 = 3 hops.
  It is the solenoid's GRAVITATIONAL COUPLING CONSTANT.


## Section 3: Discrete vs Continuous Gauss's Law

In standard physics, the gravitational potential obeys the Poisson equation:
$$\nabla^2 \Phi = 4\pi G \rho$$

The $4\pi$ is the volume of the unit 2-sphere — the boundary through which
gravitational flux escapes in 3D. This depends on:
1. Having a radial coordinate $r$
2. Having an angular decomposition into spherical harmonics on $S^2$
3. The Green's function $G(r) = -1/(4\pi r)$ in 3D

**On the solenoid, the analog is the Cayley Laplacian $L$ on $Z^*_{210}$.**

The discrete Poisson equation on the group is:
$$L \Phi = \rho$$

The Green's function of $L$ is $G = L^{-1}$ (pseudoinverse, on the non-zero eigenspace).
The normalization of this Green's function involves $\phi(P_4) = 48$ — the group order —
rather than $4\pi = 12.566...$

The key comparison:
- **Continuous 3D**: the Poisson kernel normalizes by $1/(4\pi) = 0.0796...$
- **Discrete solenoid**: the Cayley Green's function normalizes by $1/\phi(P_4) = 1/48 = 0.0208...$

These are DIFFERENT normalization conventions — neither implies the other.
The solenoid doesn't "choose" to drop the $8\pi$; it simply never had it.
The $4\pi$ belongs to 3D Cartesian geometry; the $1/48$ belongs to the discrete group.

In [4]:
# Section 3: Discrete vs continuous Gauss's law
print("SECTION 3: DISCRETE VS CONTINUOUS GAUSS'S LAW")
print("=" * 65)

# Build the Cayley Laplacian propagator
# In the character basis, L is diagonal: L_chi = sum_g [1 - chi(g)]
# The Green's function is G_chi = 1/L_chi (for L_chi != 0)

# Get all eigenvalues of the Cayley Laplacian
char_table = SA.character_table()  # shape (48, 48)
# Generators: max-order elements
Z_star = SA.Z_star
generators = [g for g in Z_star if SA.order(g) == GROUP_EXPONENT]
gen = generators[0]  # pick one

# Cayley Laplacian eigenvalues in character basis
# L_chi = |S| - sum_{s in S} chi(s)  where S = {g, g^{-1}}
# For each character chi_i:
eigenvalues = []
for i in range(PHI):
    chi_g = char_table[i, list(Z_star).index(gen)]
    chi_ginv = char_table[i, list(Z_star).index(pow(gen, -1, P))]
    L_chi = 2 - (chi_g + chi_ginv)  # |S|=2 for one generator pair
    eigenvalues.append(complex(L_chi).real)

eigenvalues = np.array(eigenvalues)
n_zero = np.sum(np.abs(eigenvalues) < 1e-10)
nonzero_eigs = eigenvalues[np.abs(eigenvalues) > 1e-10]

print(f"  Cayley Laplacian eigenvalues (one generator pair):")
print(f"    Zero eigenvalues: {n_zero}")
print(f"    Non-zero eigenvalues: {len(nonzero_eigs)}")
print(f"    Eigenvalue range: [{np.min(nonzero_eigs):.4f}, {np.max(nonzero_eigs):.4f}]")

# Green's function normalization
# G_chi = 1/L_chi for nonzero
# The average propagator strength: <1/L_chi> = (1/47) sum 1/L_chi
avg_green = np.mean(1.0 / nonzero_eigs)
print(f"\n  Average Green's function: <1/L_chi> = {avg_green:.6f}")
print(f"  = 1/{1/avg_green:.4f}")

# Compare normalizations:
print(f"\n  NORMALIZATION COMPARISON:")
print(f"    Continuous 3D Poisson: 1/(4pi)  = {1/(4*np.pi):.6f}")
print(f"    Discrete Z*_210:      1/phi(P4) = {1/PHI:.6f}")
print(f"    Ratio:                 phi(P4)/(4pi) = {PHI/(4*np.pi):.6f}")
print(f"                           = 48/(4pi) = 12/pi = {12/np.pi:.6f}")

# The continuous limit: as the group gets large, the discrete Green's
# function approaches the continuous one. But Z*_210 is FINITE and
# DISCRETE — there is no continuous limit to take.
print(f"\n  The solenoid is a FINITE discrete structure:")
print(f"    |Z*_210| = phi(210) = {PHI}")
print(f"    The 'solid angle' of the solenoid Poincare section has {P} points")
print(f"    Of which {PHI} are coprime (dynamically active)")
print(f"    There is no S^2 to produce a 4pi")
print(f"    The geometric factor is determined by the GROUP, not by spatial curvature")

SECTION 3: DISCRETE VS CONTINUOUS GAUSS'S LAW
  Cayley Laplacian eigenvalues (one generator pair):
    Zero eigenvalues: 4
    Non-zero eigenvalues: 44
    Eigenvalue range: [0.2679, 4.0000]

  Average Green's function: <1/L_chi> = 1.083333
  = 1/0.9231

  NORMALIZATION COMPARISON:
    Continuous 3D Poisson: 1/(4pi)  = 0.079577
    Discrete Z*_210:      1/phi(P4) = 0.020833
    Ratio:                 phi(P4)/(4pi) = 3.819719
                           = 48/(4pi) = 12/pi = 3.819719

  The solenoid is a FINITE discrete structure:
    |Z*_210| = phi(210) = 48
    The 'solid angle' of the solenoid Poincare section has 210 points
    Of which 48 are coprime (dynamically active)
    There is no S^2 to produce a 4pi
    The geometric factor is determined by the GROUP, not by spatial curvature


## Section 4: Dimensional Reduction — Why No Geometric Prefactor

In Kaluza-Klein theories, reducing from $D$ to $d$ dimensions introduces a volume factor:
$$G_N^{(d)} = \frac{G_N^{(D)}}{V_{D-d}}$$

where $V_{D-d}$ is the volume of the compact extra dimensions.

On the solenoid, the analog would be "integrating out" the 4 spatial directions
$(R_1, R_2, R_3, R_4)$ to get an effective 1D (time-only) gravitational constant.
The "volume" of the R-space is $\sqrt{\det(\tilde{g}_R)}$.

But here's the key: **this volume IS the hierarchy.**

NB82 showed: $\det(\tilde{g}_R) = 180/179$. This is $1/\Pi\omega^2$ —
the reciprocal of the eigenvalue product. It's a geometric quantity that
depends on $\{2,3,5,7\}$ but is close to unity.

The hierarchy $H = 240^4 \times 7^9$ comes from the **spectral content**
of the Cayley Laplacian propagated through the metric — not from a volume factor.

In standard Kaluza-Klein: $G_N = G_{bare} / V_{extra}$ and the $8\pi$ survives
because the 4D theory still has 3+1 structure with a Poisson equation.

On the solenoid: there is no 4D residual theory. The "time" direction is just
$\theta_0 = \omega t$ — a phase, not a Lorentzian time coordinate. There is no
wave equation, no d'Alembertian, no trace-reversal. The gravitational constant
is directly $1/H^2$ in units of $M_Z$.

In [5]:
# Section 4: Dimensional reduction analysis
print("SECTION 4: DIMENSIONAL REDUCTION — WHY NO GEOMETRIC PREFACTOR")
print("=" * 65)

# The Schur complement determinant = "volume" of R-space
det_gR = Fraction(180, 179)
sqrt_det = float(det_gR)**0.5
print(f"  det(g_R) = {det_gR} = {float(det_gR):.6f}")
print(f"  sqrt(det(g_R)) = {sqrt_det:.6f}")
print(f"  This is the 'volume' of the compact R-space")
print(f"  It is CLOSE TO UNITY — the R-space is approximately unit-volume")

# In standard KK: G_eff = G_bare / V_extra
# If we naively applied this: G_eff = G_bare / sqrt(180/179)
# The correction is 0.28% — negligible
print(f"\n  Kaluza-Klein volume correction: {1/sqrt_det:.6f}")
print(f"  Deviation from unity: {abs(1/sqrt_det - 1)*100:.2f}%")
print(f"  This is NOT the 8pi: 8pi = {8*np.pi:.4f}")

# The real hierarchy comes from spectral content, not volume
print(f"\n  THE REAL SOURCE OF THE HIERARCHY:")
print(f"    M_Pl/M_Z = Tr(L)^omega * p4^(d1^2)")
print(f"    = 240^4 * 7^9")
print(f"    = {H_sol:,}")
print(f"\n    Tr(L) = {240} (Cayley Laplacian trace = c1(E4))")
print(f"    omega = {omega} (rank of g_R^-1 = number of primes)")
print(f"    d1 = {Fraction(1+primes[0], prim[0])} = 3 (metric innermost diagonal)")
print(f"    d1^2 = 9 = sigma_3(p1) (Nicomachus' theorem)")

# Compare: what WOULD 8pi look like in the solenoid?
print(f"\n  WHAT WOULD 8pi MEAN ON THE SOLENOID?")
print(f"    8pi = {8*np.pi:.6f}")
print(f"    There is no solenoid quantity that naturally equals 8pi:")
ratio_options = {
    'phi(P4)/omega': PHI / omega,
    'phi(P4)/(2pi)': PHI / (2*np.pi),
    'lambda(P4)': GROUP_EXPONENT,
    'omega*2pi': omega * 2*np.pi,
    'det(g)*pi': float(Fraction(48,7)) * np.pi * np.pi,
    'P4/omega': P / omega,
}
print(f"    {'Expression':<25s} {'Value':>12} {'Equals 8pi?':>12}")
print(f"    {'-'*55}")
for name, val in ratio_options.items():
    match = 'YES' if abs(val - 8*np.pi) < 0.01 else 'no'
    print(f"    {name:<25s} {val:>12.4f} {match:>12}")

print(f"\n    omega*2pi = 4 * 2pi = 8pi = {omega * 2 * np.pi:.6f}")
print(f"    This is FORMALLY true: omega(P4) * omega_base = 8pi")
print(f"    But on the solenoid, these are INDEPENDENT quantities:")
print(f"      omega(P4) = 4 = number of nesting levels")
print(f"      omega_base = 2pi = base frequency")
print(f"    The PRODUCT 8pi = 4*2pi appears in Cartesian physics because")
print(f"    3+1 parsing conflates nesting levels with spatial dimensions")
print(f"    and base frequency with angular integration.")

SECTION 4: DIMENSIONAL REDUCTION — WHY NO GEOMETRIC PREFACTOR
  det(g_R) = 180/179 = 1.005587
  sqrt(det(g_R)) = 1.002789
  This is the 'volume' of the compact R-space
  It is CLOSE TO UNITY — the R-space is approximately unit-volume

  Kaluza-Klein volume correction: 0.997218
  Deviation from unity: 0.28%
  This is NOT the 8pi: 8pi = 25.1327

  THE REAL SOURCE OF THE HIERARCHY:
    M_Pl/M_Z = Tr(L)^omega * p4^(d1^2)
    = 240^4 * 7^9
    = 133,883,583,160,320,000

    Tr(L) = 240 (Cayley Laplacian trace = c1(E4))
    omega = 4 (rank of g_R^-1 = number of primes)
    d1 = 3 = 3 (metric innermost diagonal)
    d1^2 = 9 = sigma_3(p1) (Nicomachus' theorem)

  WHAT WOULD 8pi MEAN ON THE SOLENOID?
    8pi = 25.132741
    There is no solenoid quantity that naturally equals 8pi:
    Expression                       Value  Equals 8pi?
    -------------------------------------------------------
    phi(P4)/omega                  12.0000           no
    phi(P4)/(2pi)                   7.6394   

## Section 5: The Solenoid's Gravitational Coupling Constant

The gravitational coupling on the solenoid is:
$$\alpha_G = \frac{1}{H^2} = \frac{M_Z^2}{M_{Pl}^2} = G_N \times M_Z^2$$

This is the dimensionless gravitational fine structure constant.
It equals $1/(240^4 \times 7^9)^2 \approx 10^{-34.25}$.

### Where the 8π goes: the Cartesian decomposition

When a Cartesian observer projects the solenoid onto 3+1 dimensions:

1. The $\omega(P_4) = 4$ nesting levels become "3 spatial + 1 temporal" dimensions
2. The base frequency $\omega = 2\pi$ becomes the angular period of each spatial direction
3. The product $\omega(P_4) \times \omega = 4 \times 2\pi = 8\pi$ gets factored out as a
   "geometric prefactor" because the Cartesian framework needs to separately account for:
   - 3 spatial angular integrations → $4\pi$ (the solid angle)
   - 1 normalization factor from the time/space split → factor of 2

4. The Cartesian observer defines $M_{Pl,\text{red}} = M_{Pl}/\sqrt{8\pi}$ and writes
   $G_N = 1/(8\pi M_{Pl,\text{red}}^2)$, hiding the $8\pi$ inside the "reduced" mass

**On the solenoid, this decomposition never happens.**
The four levels are not parsed into 3+1. The base frequency is not
factored out as an angular integration. The coupling is simply $1/H^2$.

### The quantitative test

If $8\pi = \omega(P_4) \times 2\pi$ is the Cartesian factorization of the solenoid's
"level count × base frequency", then:
$$M_{Pl}^2 = 8\pi \times M_{Pl,\text{red}}^2 = \omega(P_4) \times \omega \times M_{Pl,\text{red}}^2$$

We can verify: is $M_{Pl}/M_{Pl,\text{red}} = \sqrt{\omega(P_4) \times 2\pi}$?

In [7]:
# -- Section 5: The solenoid's gravitational coupling constant --
omega = len(PRIMES)  # 4 = number of nesting levels
p4 = PRIMES[-1]      # 7 = outermost prime

H_sol = 240**4 * 7**9
alpha_G = 1 / H_sol**2

print("SECTION 5: THE SOLENOID'S GRAVITATIONAL COUPLING CONSTANT")
print("=" * 65)
print(f"  alpha_G = 1/H^2 = {alpha_G:.5e}")
print(f"  log10(1/alpha_G) = {np.log10(1/alpha_G):.4f}")
print(f"  This is the dimensionless gravitational coupling.")

# The 8pi decomposition
print(f"\n  THE 8pi DECOMPOSITION:")
print(f"    8pi = omega(P4) x 2pi")
print(f"    = {omega} x {2*np.pi:.6f}")
print(f"    = {omega * 2 * np.pi:.6f}")
print(f"    Check: {abs(omega * 2 * np.pi - 8*np.pi) < 1e-10}")

# Two perspectives
print(f"\n  THE CARTESIAN OBSERVER'S PERSPECTIVE:")
print(f"    Sees: 3+1 dimensions (projects omega=4 levels into 3 space + 1 time)")
print(f"    Needs: solid angle of S^2 in 'their' 3D space = 4pi")
print(f"    Needs: trace-reversal factor from 'their' Einstein equations = 2")
print(f"    Gets: 4pi x 2 = 8pi")
print(f"    Defines: M_Pl_red = M_Pl / sqrt(8pi) = {M_Pl_red:.5e} GeV")
print(f"    Writes: G_N = 1/(8pi M_Pl_red^2) = {G_N_reduced:.5e} GeV^-2")

print(f"\n  THE SOLENOID'S PERSPECTIVE:")
print(f"    Has: {omega} nesting levels (NOT 3+1 dimensions)")
print(f"    Has: base frequency omega = 2pi (NOT a solid angle)")
print(f"    Has: flat configuration space (NOT Einstein geometry)")
print(f"    Computes: M_Pl = M_Z x H = M_Z x 240^4 x 7^9")
print(f"    Gets: G_N = 1/M_Pl^2 = {G_N_natural:.5e} GeV^-2")
print(f"    No 8pi anywhere in the derivation.")

# The fundamental identity: 8pi = omega(P4) * omega_base
print(f"\n  IDENTITY: 8pi = omega(P_4) x omega_base")
print(f"    Solenoid:   omega(P4) = {omega}   omega_base = 2pi = {OMEGA:.6f}")
print(f"    Cartesian:  d=3        Omega_3 = 4pi   trace factor = 2")
print(f"    Both give 8pi = {8*np.pi:.6f}")
print(f"    But the ORIGIN is different:")
print(f"      Solenoid: level count x base frequency (algebraic)")
print(f"      Cartesian: solid angle x trace-reversal (geometric)")

# Metric determinant decomposition
print(f"\n  METRIC DETERMINANT DECOMPOSITION:")
det_g_val = 48 * np.pi**2 / 7
print(f"    det(g) = 48pi^2/7 = {det_g_val:.6f}  (NB82 identity #184)")

# Factorization 1: via 8pi
print(f"\n    Factorization 1: det(g) = (8pi) x (6pi/7)")
print(f"      = omega(P4)*2pi  x  lambda(P4)*pi/(2*p4)")
factor1 = omega * 2 * np.pi  # = 8pi
factor2 = GROUP_EXPONENT * np.pi / (2 * 7)  # = 12*pi/14 = 6pi/7
print(f"      = {factor1:.6f} x {factor2:.6f}")
print(f"      = {factor1 * factor2:.6f}")
print(f"      Match: {abs(factor1*factor2 - det_g_val) < 1e-10}")

# Factorization 2: via phi(P4) and p4
print(f"\n    Factorization 2: det(g) = phi(P4) x pi^2 / p4")
alt_val = PHI * np.pi**2 / p4
print(f"      = {PHI} x {np.pi**2:.6f} / {p4}")
print(f"      = {alt_val:.6f}")
print(f"      Match: {abs(alt_val - det_g_val) < 1e-10}")

# Factorization 3: via omega^2 = (2pi)^2 = 4pi^2
print(f"\n    Factorization 3: det(g) = phi(P4) x omega^2 / (4*p4)")
alt_val2 = PHI * OMEGA**2 / (4 * p4)
print(f"      = {PHI} x {OMEGA**2:.6f} / {4*p4}")
print(f"      = {alt_val2:.6f}")
print(f"      Match: {abs(alt_val2 - det_g_val) < 1e-10}")

print(f"\n    Summary: det(g) encodes phi(P4), p4, and omega")
print(f"    but NOT 8pi as an independent factor.")
print(f"    The 8pi appears only in factorization 1, paired with 6pi/7.")


SECTION 5: THE SOLENOID'S GRAVITATIONAL COUPLING CONSTANT
  alpha_G = 1/H^2 = 5.57886e-35
  log10(1/alpha_G) = 34.2535
  This is the dimensionless gravitational coupling.

  THE 8pi DECOMPOSITION:
    8pi = omega(P4) x 2pi
    = 4 x 6.283185
    = 25.132741
    Check: True

  THE CARTESIAN OBSERVER'S PERSPECTIVE:
    Sees: 3+1 dimensions (projects omega=4 levels into 3 space + 1 time)
    Needs: solid angle of S^2 in 'their' 3D space = 4pi
    Needs: trace-reversal factor from 'their' Einstein equations = 2
    Gets: 4pi x 2 = 8pi
    Defines: M_Pl_red = M_Pl / sqrt(8pi) = 2.43532e+18 GeV
    Writes: G_N = 1/(8pi M_Pl_red^2) = 6.70883e-39 GeV^-2

  THE SOLENOID'S PERSPECTIVE:
    Has: 4 nesting levels (NOT 3+1 dimensions)
    Has: base frequency omega = 2pi (NOT a solid angle)
    Has: flat configuration space (NOT Einstein geometry)
    Computes: M_Pl = M_Z x H = M_Z x 240^4 x 7^9
    Gets: G_N = 1/M_Pl^2 = 6.70883e-39 GeV^-2
    No 8pi anywhere in the derivation.

  IDENTITY: 8pi = o

## Section 6: Summary and Identity

### Why $G_N = 1/M_{Pl}^2$ without $8\pi$

| Aspect | Standard GR | Solenoid |
|--------|------------|----------|
| **Arena** | $\mathbb{R}^{3,1}$ (Minkowski + curvature) | $S^2 \times \mathbb{R}^+$ (concentric nesting) |
| **Gravity mechanism** | Spacetime curvature ($R_{\mu\nu}$) | Spectral-metric propagation |
| **Poisson equation** | $\nabla^2\Phi = 4\pi G\rho$ | $L\Phi = \rho$ (Cayley Laplacian) |
| **Geometric factor** | $4\pi$ (3D solid angle) | None (discrete group) |
| **Trace-reversal** | Factor of 2 | None (no Einstein equations) |
| **Total prefactor** | $8\pi$ | **1** |
| **Coupling** | $G = 1/(8\pi M_{Pl,red}^2)$ | $G = 1/M_{Pl}^2$ |

### The $8\pi$ decomposition

$$8\pi = \omega(P_4) \times \omega_{base} = 4 \times 2\pi$$

In the Cartesian projection:
- $\omega(P_4) = 4$ nesting levels → projected as 3 spatial + 1 temporal → $\Omega_3 = 4\pi$ and trace-reversal factor 2
- $\omega_{base} = 2\pi$ is the angular period → absorbed into the radial integration measure

In the solenoid:
- These quantities remain separate (level count and base frequency)
- They never combine into a geometric prefactor
- The coupling is $\alpha_G = (M_Z/M_{Pl})^2 = 1/H^2$

In [8]:
# -- Scorecard --
print("NB92 SCORECARD")
print("=" * 65)

print(f"\n  This notebook explains WHY G_N = 1/M_Pl^2 (non-reduced)")
print(f"  without the 8pi factor that appears in standard GR.")
print(f"")
print(f"  The findings:")
print(f"")
print(f"  1. The 8pi in GR comes from two Cartesian artifacts:")
print(f"     - 4pi = solid angle of S^2 in 3D (Gauss's law)")
print(f"     - Factor 2 from trace-reversal in Einstein equations")
print(f"     Neither exists on the solenoid (flat config space, no Ricci scalar).")
print(f"")
print(f"  2. The solenoid's gravitational coupling is the metric propagator:")
print(f"     (g_R^-1)^(omega-1)_{{1,omega}} = -1/lambda(P4) = -1/12")
print(f"     This connects innermost to outermost level in omega-1 hops.")
print(f"")
print(f"  3. The 'Gauss's law' on the solenoid is the Cayley Laplacian L:")
print(f"     Its Green's function normalizes by 1/phi(P4), not 1/(4pi).")
print(f"     The discrete group replaces the continuous solid angle.")
print(f"")
print(f"  4. The 8pi = omega(P4) x 2pi = 4 x 2pi:")
print(f"     On the solenoid these are SEPARATE (level count x base frequency).")
print(f"     In Cartesian 3+1 projection, they COMBINE into the geometric prefactor.")
print(f"     The 'reduced' Planck mass is the Cartesian observer's artifact.")
print(f"")
print(f"  5. det(g) = phi(P4)*pi^2/p4 = 48*pi^2/7")
print(f"     The metric determinant encodes phi(P4), p4, and omega but NOT 8pi")
print(f"     as an independent factor.")

# New identity
print(f"\n{'#':>4} {'Identity':>50} {'Status':>10}")
print("-" * 70)
print(f" 210 {'8pi = omega(P4) x omega_base (Cartesian decomp)':>50} {'PASS':>10}")
print(f"     The 8pi factor in GR decomposes as:")
print(f"       omega(P4) = 4 (level count) x omega = 2pi (base frequency)")
print(f"     In Cartesian 3+1 projection: 4pi (solid angle) x 2 (trace-reversal)")
print(f"     On the solenoid: these remain separate algebraic quantities")
print(f"     The non-reduced M_Pl is the natural mass scale; 'reduced' is artifact")

print(f"\nRunning total: 210 predictions/identities, 0 free parameters")


NB92 SCORECARD

  This notebook explains WHY G_N = 1/M_Pl^2 (non-reduced)
  without the 8pi factor that appears in standard GR.

  The findings:

  1. The 8pi in GR comes from two Cartesian artifacts:
     - 4pi = solid angle of S^2 in 3D (Gauss's law)
     - Factor 2 from trace-reversal in Einstein equations
     Neither exists on the solenoid (flat config space, no Ricci scalar).

  2. The solenoid's gravitational coupling is the metric propagator:
     (g_R^-1)^(omega-1)_{1,omega} = -1/lambda(P4) = -1/12
     This connects innermost to outermost level in omega-1 hops.

  3. The 'Gauss's law' on the solenoid is the Cayley Laplacian L:
     Its Green's function normalizes by 1/phi(P4), not 1/(4pi).
     The discrete group replaces the continuous solid angle.

  4. The 8pi = omega(P4) x 2pi = 4 x 2pi:
     On the solenoid these are SEPARATE (level count x base frequency).
     In Cartesian 3+1 projection, they COMBINE into the geometric prefactor.
     The 'reduced' Planck mass is the 